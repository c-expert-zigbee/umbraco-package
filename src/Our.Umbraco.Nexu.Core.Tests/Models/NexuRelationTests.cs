namespace Our.Umbraco.Nexu.Core.Tests.Models
{
    using System;

    using ApprovalTests;
    using ApprovalTests.Reporters;
    using ApprovalTests.Reporters.ContinuousIntegration;

    using Newtonsoft.Json;

    using NUnit.Framework;

    using Our.Umbraco.Nexu.Common.Models;

    /// <summary>
    /// Represents nexu relation model tests.
    /// </summary>
    [TestFixture]    
    [UseReporter(typeof(DiffReporter), typeof(AppVeyorReporter))]
    public class NexuRelationTests
    {
        [Test]
        public void When_NexuRelation_Instance_Is_Created_Id_Should_Be_A_Guid()
        {
            // act
            var result = new NexuRelation();

            // assert
            Assert.That(result.Id != Guid.Empty);
        }

        [Test]
        public void Nexu_Relation_Udi_Should_Match_Id()
        {
            // act
            var result = new NexuRelation
                             {
                                 Id= new Guid("9a935375-6c89-4dda-ae83-b9207e19a3ee")
                             };

            // assert
            Approvals.VerifyJson(JsonConvert.SerializeObject(result));           
        }
    }
}
