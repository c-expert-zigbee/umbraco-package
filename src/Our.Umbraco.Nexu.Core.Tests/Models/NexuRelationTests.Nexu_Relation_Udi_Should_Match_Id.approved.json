{
    "Id": "9a935375-6c89-4dda-ae83-b9207e19a3ee",
    "ParentUdi": null,
    "ChildUdi": null,
    "RelationType": "00000000-0000-0000-0000-000000000000",
    "PropertyAlias": null,
    "Culture": null,
    "Udi": "umb://nexurelation/9a9353756c894ddaae83b9207e19a3ee"
}